core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -267,10 +292,22 @@ object SparkSubmit {
     // In yarn-cluster mode, use yarn.Client as a wrapper around the user class
     if (isYarnCluster) {
       childMainClass = "org.apache.spark.deploy.yarn.Client"
-      if (args.primaryResource != SPARK_INTERNAL) {
-        childArgs += ("--jar", args.primaryResource)
+      if (args.isPython) {
+        val mainPyFile = new Path(args.primaryResource).getName
+        childArgs += ("--primary-py-file", mainPyFile)
+        val pyFilesNames: String = if (args.pyFiles != null) {
+          args.pyFiles.split(",").map { p => (new Path(p)).getName }.mkString(",")
```

Comment:
Use parentheses instead of curly braces here.  Also, probably better to use java.io.File than Path, because Hadoop isn't involved and the file is local.